### PR TITLE
Resolve menu item with no locale

### PR DESCRIPTION
**Pull request by mariavilaro**
I moved this change to its own branch, please close #199

This is a change that, along with my other pull request in the [sitemap plugin](https://github.com/rainlab/sitemap-plugin) allows to generate multilanguage sitemaps, by generating a sitemap.xml with all the urls in all languages.

### CODE REVIEW

File: formwidgets/mltext/partials/_mltext.htm [hunk 1]
```diff
@@ -17,7 +17,7 @@
             placeholder="<?= e(trans($field->placeholder)) ?>"
             class="form-control"
             autocomplete="off"
-            maxlength="255"
+            <?= $field->hasAttribute('maxlength') ? '' : 'maxlength="255"' ?>
```
Comment (mjauvin) [inline formwidgets/mltext/partials/_mltext.htm]: This change is not related to the current PR, can you create an independent PR if you want to fix this?

**Comment by mjauvin** [comment]
Can you reference the related PR in the Sitemap plugin here?

File: behaviors/TranslatablePageUrl.php [hunk 2]
```diff
@@ -172,4 +173,34 @@ public function setViewBagUrlAttributeTranslated($value, $locale)
...
+    public static function resolveMenuItem($item, $url, $theme)
+    {
+        $result = null;
+
+        if ($item->type == 'cms-page') {
+            if (!$item->reference) {
+                return;
+            }
+
+            $page = \Cms\Classes\Page::loadCached($theme, $item->reference);
```
Comment (mjauvin) [inline behaviors/TranslatablePageUrl.php]: Please, include this Class instead of referencing the full namespace here.

File: behaviors/TranslatablePageUrl.php [hunk 3]
```diff
@@ -172,4 +173,34 @@ public function setViewBagUrlAttributeTranslated($value, $locale)
...
+            $translate = Translator::instance();
+            if ($page->hasTranslatablePageUrl($translate->getLocale())) {
+                $page->rewriteTranslatablePageUrl($translate->getLocale());
+            }
+            $pageUrl = $translate->getPathInLocale($page->url, $translate->getLocale());
```
Comment (mjauvin) [inline behaviors/TranslatablePageUrl.php]: You're using "translate->getLocale()" 3 times in a row... please, assign the result to a variable and reuse variable.

**Comment by mjauvin** [comment]
Closing as it has been over a month since any activity on this occurred and we are trying to figure out what issues are still relevant. If this is still something that you would like to see through to fruition please respond and we can get the ball rolling.

**Comment by mariavilaro** [comment]
Hi, sorry, no this is no longer needed
